pkg/fqdn/cache.go

```diff
@@ -268,6 +323,7 @@ func (c *DNSCache) updateWithEntryIPs(entries ipEntries, entry *cacheEntry) {
 			c.upsertReverse(ipStr, entry)
 		}
 	}
+	c.addNameToCleanup(entry)
```

Comment:
This needs to be called when we call `c.upsertReverse` (just above) otherwise we introduce expirations that aren't actually happening in the cache.